app/(candidate)/dashboard/contacts/[[...attr]]/components/Download.js

```diff
@@ -5,11 +5,19 @@ import { fetchContactsCsv } from './shared/ajaxActions'
 import { dateUsHelper } from 'helpers/dateHelper'
 import { EVENTS, trackEvent } from 'helpers/analyticsHelper'
 import { isCustomSegment, findCustomSegment } from './shared/segments.util'
+import { useCampaign } from '@shared/hooks/useCampaign'
+import { useShowContactProModal } from '../hooks/ContactProModal'
 
 export default function Download() {
+  const [campaign] = useCampaign()
+  const showProUpgradeModal = useShowContactProModal()
   const [customSegments, , , querySegment] = useCustomSegments()
 
   const handleDownload = async () => {
+    if (!campaign.isPro) {
+      showProUpgradeModal(true)
+      return
+    }
```

Comment:
Not blocking. But this is rather repetitive.. could make future refactors risky/tough

Maybe find a way to abstract this behavior into a single place? 🙏

Reply:
Yeah...I thought about this a while, and I couldn't think of anything cleaner. I considered these alternatives:
- Maintain dedicated state + modals in each sub-component (this seemed worse)
- Add a `onShouldShowProModal` prop to each of these components, and then call up in each case. The tradeoffs here seemed really similar in to the current approach, except the current approach results in less boilerplate.

Any other ideas? I came up dry.

Reply:
Hmm.. what about a hook that accepts the event handler as an argument. Then, if it should just show the modal, it returns that as the event handler, but if not, returns the argument handler?

I dunno.. just spit ballin'.. Like I said, not blocking.. just an idea.